commercetools.NET/Common/Client.cs

```diff
@@ -277,23 +317,33 @@ public Client(Configuration configuration)
 
             string credentials = Convert.ToBase64String(Encoding.ASCII.GetBytes(string.Concat(this.Configuration.ClientID, ":", this.Configuration.ClientSecret)));
 
-            using (HttpClient client = new HttpClient(new ClientLoggingHandler(new HttpClientHandler())))
+            for (int internalServerErrorRetries = -1; internalServerErrorRetries < this.Configuration.InternalServerErrorRetries; internalServerErrorRetries++)
             {
-                client.DefaultRequestHeaders.Accept.Clear();
-                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", credentials);
-                client.DefaultRequestHeaders.UserAgent.ParseAdd(this.UserAgent);
-
-                HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, new Uri(this.Configuration.OAuthUrl))
+                using (HttpClient client = new HttpClient(new ClientLoggingHandler(new HttpClientHandler())))
                 {
-                    Version = HttpVersion.Version10,
-                    Content = new FormUrlEncodedContent(pairs)
-                };
+                    client.DefaultRequestHeaders.Accept.Clear();
+                    client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", credentials);
+                    client.DefaultRequestHeaders.UserAgent.ParseAdd(this.UserAgent);
 
-                HttpResponseMessage httpResponseMessage = await client.SendAsync(httpRequestMessage);
-                response = await GetResponse<Token>(httpResponseMessage);
-            }
+                    HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, new Uri(this.Configuration.OAuthUrl))
+                    {
+                        Version = HttpVersion.Version10,
+                        Content = new FormUrlEncodedContent(pairs)
+                    };
 
+                    HttpResponseMessage httpResponseMessage = await client.SendAsync(httpRequestMessage);
+                    response = await GetResponse<Token>(httpResponseMessage);
+                }
+                if (response.StatusCode < 500)
+                {
+                    return response;
+                }
+                else if (this.Configuration.InternalServerErrorRetryInterval > 0)
```

Comment:
This check should be done only once during construction of the Configuration object.

Reply:
No. As the application may run longer than the token life time the refreshToken task may be called multiple times.